android/src/main/java/com/reactnativestripesdk/PaymentLauncherFragment.kt

```diff
@@ -0,0 +1,115 @@
+package com.reactnativestripesdk
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.fragment.app.Fragment
+import com.facebook.react.bridge.Promise
+import com.stripe.android.ApiResultCallback
+import com.stripe.android.Stripe
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.payments.paymentlauncher.PaymentLauncher
+import com.stripe.android.payments.paymentlauncher.PaymentResult
+
+class PaymentLauncherFragment(
+  private val stripe: Stripe,
+  private val publishableKey: String,
+  private val stripeAccountId: String?,
+) : Fragment() {
+  lateinit var paymentLauncher: PaymentLauncher
+
+  var clientSecret: String? = null
+  var promise: Promise? = null
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View {
+    paymentLauncher = createPaymentLauncher()
+    return FrameLayout(requireActivity()).also {
+      it.visibility = View.GONE
+    }
+  }
+
+  private fun createPaymentLauncher(): PaymentLauncher {
+    return PaymentLauncher.create(this, publishableKey, stripeAccountId) { paymentResult ->
+      when (paymentResult) {
+        is PaymentResult.Completed -> {
+          clientSecret?.let {
+            retrievePaymentIntent(it, stripeAccountId)
```

Comment:
How is the result of of retrievePaymentIntent being used?

Reply:
`retrievePaymentIntent` allows us to get the relevant `paymentIntent` (or the error) and pass it back to the React Native app via `promise.resolve`. Went with `stripe.retrievePaymentIntent` based on [this comment](https://github.com/stripe/stripe-android/issues/2640#issuecomment-929682178)

Reply:
I am not sure I am understanding.  It looks like the retrievePaymentIntent is only be retrieved in the "Complete" case and isn't being returned.

Reply:
That's correct- `retrievePaymentIntent` is only called in the complete case. In the case of cancellation or failure, we send an error object back to the user (L46 and L50)

In the case of a completed payment result, we retrieve the payment intent, and `resolve` the promise that was passed in via Javascript. We aren't explicitly calling `return` here with the payment intent object, but the `promise.resolve` call is passing the data back over the React Native bridge into "Javascript land". 

The previous code functioned the same way, [see here](https://github.com/stripe/stripe-react-native/blob/master/android/src/main/java/com/reactnativestripesdk/StripeSdkModule.kt#L82-L135). But, if there's a better way to fetch the payment intent associated with the result then let me know!

Reply:
`PaymentLauncher` actually [processes](https://github.com/stripe/stripe-android/blob/262885d0c02a54430ae8f0c3966de694abc6bb0a/payments-core/src/main/java/com/stripe/android/StripeIntentResult.kt#L33) the status of the Intent internally and [returns](https://github.com/stripe/stripe-android/blob/d8460ee2f2c554df245cf7a0ca92d76be6195bca/payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncherViewModel.kt#L222) `Completed`/`Failed`/`Cancelled` according to the processed `Outcome`s.

Compared the logic with `retrievePaymentIntent` a bit, it seems we can just save the retrievePaymentIntent all and do the following mapping:
```
PaymentResult.Completed-> {
  val paymentIntent = createResult("paymentIntent", mapFromPaymentIntentResult(result))
  promise.resolve(paymentIntent)}
PaymentResult.Failed-> {
  val error = result.lastPaymentError
  promise.resolve(createError(ConfirmPaymentErrorType.Failed.toString(), error))
}
PaymentResult.Cancelled-> {
if (isPaymentIntentNextActionVoucherBased(result.nextActionType)) {
    val paymentIntent = createResult("paymentIntent", mapFromPaymentIntentResult(result))
    promise.resolve(paymentIntent)
  } else {
    (result.lastPaymentError)?.let {
      promise.resolve(createError(ConfirmPaymentErrorType.Canceled.toString(), it))
    } ?: run {
      promise.resolve(createError(ConfirmPaymentErrorType.Canceled.toString(), "The payment has been canceled"))
    }
  }
}
```

Reply:
Not sure I understand- `mapFromPaymentIntentResult` accepts a `PaymentIntent`, but the `result` we're getting here from `PaymentLauncher` is a `PaymentResult`. I thought in order to retrieve the full `PaymentIntent` object, we pretty much needed to use `Stripe.retrievePaymentIntent`